app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -24,37 +24,55 @@ class PromotedStoryListView @JvmOverloads constructor(
 
   @Inject
   lateinit var bindingInterface: ViewBindingShim
+  private var dataList: List<PromotedStoryViewModel> = listOf()
 
   override fun onAttachedToWindow() {
     super.onAttachedToWindow()
 
     (FragmentManager.findFragment<Fragment>(this) as ViewComponentFactory)
       .createViewComponent(this).inject(this)
 
-    adapter = BindableAdapter.SingleTypeBuilder.newBuilder<PromotedStoryViewModel>()
+    /**
+     * The StartSnapHelper is used to snap between items rather than smooth scrolling, so that
+     * the item is completely visible in [HomeFragment] as soon as learner lifts the finger
+     * after scrolling.
+     */
+    val snapHelper = StartSnapHelper()
+    this.onFlingListener = null
```

Comment:
Nit: ``this.`` is unneeded here since it's unambiguous.

Reply:
Done